Bio/SeqUtils/__init__.py

```diff
@@ -467,6 +468,69 @@ def six_frame_translations(seq, genetic_code=1):
     return res
 
 
+def count_kmers(seq, k, circular=False, ordered=True):
+    """Count how many times each kmer appears in seq.
+
+    Counts instances of each substring of size k (kmers) in seq. Returns a
+    python collections.Counter object, with the kmers as keys and the number
+    of occurence as values. Overlaping k-mers, such as "ARA" in "ARARA" will be
+    counted twice.
+
+    Arguments:
+     - seq: String or Biopython sequence object.
+     - k: integer with size of kmer to be counted.
+     - circular: bool. Is the molecule circular?
+     - ordered: bool. If False will count "AGC" and "CGA" as the same kmer. Will
+                also count palindromes twice.
```

Comment:
What the code does appears to be counting the kmer and its reversed string - not ordered in the sense of sorting the letters which what the argument name suggested. What is this for?

(See also comment about the implementation and speed below)